### PR TITLE
Add obscureAnnotation / removeAnnotation to InTextAnnotationParser, r…

### PR DESCRIPTION
**Pull request by mwjames**
…efs #1314

- `InTextAnnotationParser::removeAnnotation` to entirely remove traces of any annotation within the text
- `InTextAnnotationParser::obscureAnnotation` to encode part of the annotation so that
the `InTextAnnotationParser::parse` is not able to process that part of a text

### CODE REVIEW

**Comment by mwjames** [comment]
`InTextAnnotationParser::obscureAnnotation` to produce

image

`InTextAnnotationParser::removeAnnotation` to produce

image

File: src/InTextAnnotationParser.php [hunk 1]
```diff
@@ -141,6 +141,64 @@ public function parse( &$text ) {
 	}
 
 	/**
+	 * @since 2.4
+	 *
+	 * @param string $text
+	 *
+	 * @return text
+	 */
+	public static function obscureAnnotation( $text ) {
```
Comment (JeroenDeDauw) [inline src/InTextAnnotationParser.php]: 3326873_orig

Production usage of this new method is not in this PR, so it's hard to say much more about this

File: src/InTextAnnotationParser.php [hunk 2]
```diff
@@ -141,6 +141,64 @@ public function parse( &$text ) {
...
+
+		// Convert possible internal URL encoding
+		$text = str_replace( array( '%5B', '%5D' ), array( '[', ']' ), $text );
```
Comment (JeroenDeDauw) [inline src/InTextAnnotationParser.php]: You don't need this local var, or the override the previous value of the parameter. You can just inline this below. At the same time it seems like a good candidate for extraction into its own method, due to the comment and the presence of low level details.

File: src/InTextAnnotationParser.php (hunk 1 above)
Comment (mwjames) [inline src/InTextAnnotationParser.php]: Replaces [0].

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/src/Error.php#L63-L64

**Pull request merged**